Source/monster.cpp

```diff
@@ -1467,7 +1467,7 @@ bool MonsterRangedSpecialAttack(int monsterId)
 	assert(monsterId >= 0 && monsterId < MaxMonsters);
 	auto &monster = Monsters[monsterId];
 
-	if (monster.AnimInfo.currentFrame == monster.data().mAFNum2 - 1 && monster.AnimInfo.tickCounterOfCurrentFrame == 0) {
+	if (monster.AnimInfo.currentFrame == monster.data().mAFNum2 - 1 && monster.AnimInfo.tickCounterOfCurrentFrame == 0 && monster._mVar2 == 0) {

```

Comment:
What does var2 do here? Can we trust it wasn't set for some odd reason that we didn't think off?

Reply:
- For `_mVar2` Mega-AI it [counts the game ticks](https://github.com/diasurgical/devilutionX/blob/7612fb4bb5636f993afbad9e15fffa09e204ff89/Source/monster.cpp#L1482-L1488) that the mega sits when casting inferno.
- The added check ensures that Mega-AI casts only one inferno while sitting.
- `_mVar2` is initialized in [StartRangedSpecialAttack](https://github.com/diasurgical/devilutionX/blob/7612fb4bb5636f993afbad9e15fffa09e204ff89/Source/monster.cpp#L905). 
- `_mVar2` is also used for walking, delay and normal ranged attacks. But in these cases monsters have a different `_mmode`.

If wanted i can modify the check to `&& (monster._mAi != AI_MEGA || monster._mVar2 == 0)`.
This would make it even more safe, cause in case of Mega-AI  `_mVar2` must be correct, else the sitting would not work as expected.

Reply:
```suggestion
	if (monster.AnimInfo.currentFrame == monster.data().mAFNum2 - 1 && monster.AnimInfo.tickCounterOfCurrentFrame == 0 && (monster._mAi != AI_MEGA || monster._mVar2 == 0)) {
```